hypervisor/arch/x86/irq.c

```diff
@@ -41,7 +41,7 @@ uint32_t alloc_irq_num(uint32_t req_irq)
 	if (irq == IRQ_INVALID) {
 		/* if no valid irq num given, find a free one */
 		irq = ffz64_ex(irq_alloc_bitmap, NR_IRQS);
-		irq = (irq == NR_IRQS) ? IRQ_INVALID : irq;
+		irq = (irq >= NR_IRQS) ? IRQ_INVALID : irq;
```

Comment:
Should we change the code to:
if (...)
    irq = IRQ_INVALID.

This way, in normal case, we avoid a redundant operation of irq = irq.

Reply:
can you check the entire HV to see if this kind of issues exist?